archive.tf

```diff
@@ -1,7 +1,13 @@
+locals {
+  archive_external_cmd      = "${concat(var.python_cmd, list("${path.module}/hash.py"))}"
+  archive_null_resource_cmd = "${concat(var.python_cmd, list(lookup(data.external.archive.result, "build_command")))}"
```

Comment:
This threw me off. This will be like `["python", "build.py xyz"]` which is doesn't make sense, as commands in general should be a list of arguments like `["python", "build.py", "xyz"]` or a single string like `"python build.py xyz"` . It is only when it gets joined with `join(" ", local.archive_null_resource_cmd)` that the command makes sense.

I also struggled to understand the purpose of these local variables, and had to keep referring back to where they're being used to make sense of them. I think having them up here makes it harder to follow the code.

So for those 2 reasons, please stop using these 3 local variables and move the code to where they're being used.